catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java

```diff
@@ -66,43 +64,38 @@ public SigningResponse sign(SigningRequest clientRequest) {
       throw new IllegalArgumentException("DELETE requests must have a non-empty body");
     }
 
-    S3BucketOptions bucketOptions =
-        s3Options.effectiveOptionsForBucket(clientRequest.bucket(), secretsProvider);
-    AwsCredentialsProvider credentialsProvider =
-        S3Clients.awsCredentialsProvider(bucketOptions, s3sessions);
-
     SdkHttpFullRequest.Builder request =
         SdkHttpFullRequest.builder()
             .uri(uri)
             .protocol(uri.getScheme())
             .method(SdkHttpMethod.fromValue(clientRequest.method()))
             .headers(clientRequest.headers());
 
-    if (body.isEmpty()) {
-      request.putHeader(SignerConstant.X_AMZ_CONTENT_SHA256, EMPTY_BODY_SHA256);
-    } else {
-      request.contentStreamProvider(() -> new ByteArrayInputStream(body.get().getBytes(UTF_8)));
-    }
+    S3BucketOptions bucketOptions =
+        s3Options.effectiveOptionsForBucket(clientRequest.bucket(), secretsProvider);
+    AwsCredentialsProvider credentialsProvider =
+        S3Clients.awsCredentialsProvider(bucketOptions, s3sessions);
+    AwsCredentialsIdentity credentials = credentialsProvider.resolveCredentials();
+
+    SignRequest.Builder<AwsCredentialsIdentity> signRequest =
+        SignRequest.builder(credentials)
+            .request(request.build())
+            .putProperty(AwsV4HttpSigner.REGION_NAME, clientRequest.region())
+            .putProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, S3Client.SERVICE_NAME)
+            .putProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, false)
+            .putProperty(AwsV4HttpSigner.NORMALIZE_PATH, false)
+            .putProperty(AwsV4HttpSigner.CHUNK_ENCODING_ENABLED, false)
+            .putProperty(AwsV4HttpSigner.PAYLOAD_SIGNING_ENABLED, false);
```

Comment:
Should we enforce the algorithm?
```suggestion
            .putProperty(AwsV4HttpSigner.PAYLOAD_SIGNING_ENABLED, false)
            .putProperty(AwsV4HttpSigner.CHECKSUM_ALGORITHM, DefaultChecksumAlgorithm.SHA256);
```

Reply:
That's been there before. 🤷

Reply:
And the suggested change actually breaks the functionality :(

Reply:
BTW: It uses SHA256 (I looked into the headers)